### ISSUE TITLE
memory leak on MacOS?

### ISSUE THREAD

**Issue by pjv** [keybase/kbfs#1529]
Without doing anything with it (i.e. no file operations) the kbfs process on my mac has been steadily eating more memory day by day lately.

I've been killing it and letting it re-launch when it gets to around 1 GB...

<img width="826" alt="2018-03-22 at 7 58 am" src="https://user-images.githubusercontent.com/327716/37771844-d7d5e180-2da6-11e8-8021-d3e4dff243f3.png">

**Comment by strib** [comment]
@pjv please do a `keybase log send` and we can take a look.  We did just plug a known memory leak recently, and the fix hasn't been released yet, so it might be the same thing.

**Comment by pjv** [comment]
@strib done. In the feedback i referenced this issue.

**Comment by strib** [comment]
Thanks!  Logs: 3db962a332d3e73b56f64c1c

I will look later today.

**Comment by strib** [comment]
@pjv hrm I don't see any activity in your logs, as you say, even connection issues (which were related to the latest leak we plugged).

I also don't see any restarts (the logs that were sent only cover the last day or so), so I'm not 100% sure about the state it was in before your last restart.

If the memory is still currently growing, would you be willing to share some more debugging with us please?  If so, please do:

```sh
cp /keybase/.kbfs_profiles/heap /keybase/private/strib,pjv/
cp /keybase/.kbfs_profiles/goroutine /keybase/private/strib,pjv/
tar -czf /keybase/private/strib,pjv/client-1529.tgz ~/Library/Logs/keybase.kbfs.*
```

Thanks!

**Comment by pjv** [comment]
@strib interesting, have not done anything keybase-related since sending that log earlier. Memory footprint for kbfs is half what it was then.

<img width="784" alt="2018-03-22 at 1 39 pm" src="https://user-images.githubusercontent.com/327716/37791169-7e6d779a-2dd6-11e8-98a2-c338e7cbf7c8.png">

happy to send the additional data if you still want it. Lmk.

**Comment by strib** [comment]
Ah ok cool, then maybe everything is ok.  Our code is in Go, and it has some lazy memory deallocating, maybe that is what's going on.  If you see it reach 1 GB again without using KBFS, please send the above data, otherwise I don't think there's much to look at yet.

**Comment by pjv** [comment]
k, will-do.

**Comment by mirorauhala** [comment]
<img width="1255" alt="screen shot 2018-06-08 at 2 02 19" src="https://user-images.githubusercontent.com/4082806/41130646-686de5de-6ac0-11e8-9111-813ab1107407.png">

I recently deleted a folder 50 GB worth of data and ram usage jumped through the roof. Whole system went slow and started to freeze.

**Comment by strib** [comment]
Or I guess if KBFS is too overloaded, just copy those files somewhere local and get them to me some other way (maybe after KBFS is restarted).

**Comment by mirorauhala** [comment]
I had to quit the app. I'll try relaunching it.

**Comment by strib** [comment]
Ah bummer.  Ok, I'll look at the logs.

**Comment by mirorauhala** [comment]
It's now happening again. 3gb and counting.

**Comment by strib** [comment]
Can you please send the heap and goroutine file my way, as mentioned above?

**Comment by mirorauhala** [comment]
Yup!

**Comment by strib** [comment]
Thanks!  @jzila looks like that fun prefetcher leak again.  I still have no clue about this one though.  See the attached graph.

[profile001.pdf](https://github.com/keybase/kbfs/files/2082399/profile001.pdf)

@mirorauhala sorry about this.  I've seen this before but we've never been able to reproduce it locally reliably, in order to debug it.  The second time it happened, did you first access anything in KBFS, or did it just start happening on its own after the restart?

**Comment by mirorauhala** [comment]
A long time ago uploaded a folder with a lot of data. Now I deleted the folder in question using Finder. I noticed that the laptop got really slow and started to heat up. Ram usage was 5gb. Then I quit the app from the menubar. Restarted by just launching the app by hitting `cmd-space > Keybase` and then looked at Activity Monitor. Currently it has lowered itself to 2,49GB.

**Comment by mirorauhala** [comment]
I uploaded new files if that helps.

**Comment by strib** [comment]
Ok thanks.  It might still be draining data from your local journal as it sends the delete messages to our server, and that somehow is triggering this extra memory usage.  It could be that just letting it run over night if possible, until it drains all those deletes, and then restarting the app will fix your problem.

Deleting large folders is not our strong suit, and I've seen a leak like this before, but it's very mysterious still.  Sorry for the bad experience, hopefully we can get to the bottom of this soon.

**Comment by mirorauhala** [comment]
Yeah, currently there are only two major issues holding me from using Keybase. Large file handling and selective sync.

**Comment by jzila** [comment]
For selective sync an okay substitute can be to use teams and subteams and syncing those.

**Comment by mirorauhala** [comment]
I have not yet tried teams. Will give that a shot.
